Platforms/NapaliPkg/Device/xiaomi-polaris/Library/PlatformMemoryMapLib/PlatformMemoryMapLib.c

```diff
@@ -0,0 +1,109 @@
+    { "MPSS_EFS"           ,0x85D00000, 0x00200000, AddMem, SYS_MEM, SYS_MEM_CAP, Reserv, NS_DEVICE },
+    { "GPU PRR"            ,0x85F00000, 0x00010000, AddMem, SYS_MEM, SYS_MEM_CAP, Reserv, NS_DEVICE },
+    { "HLOS2"              ,0x85F10000, 0x000D0000, AddMem, SYS_MEM, SYS_MEM_CAP, BsData, WRITE_BACK_XN },
+    { "AOP CMD DB"         ,0x85FE0000, 0x00020000, AddMem, SYS_MEM, SYS_MEM_CAP, Reserv, WRITE_BACK_XN },
+    { "SMEM"               ,0x86000000, 0x00200000, AddMem, MEM_RES, UNCACHEABLE, Reserv, UNCACHED_UNBUFFERED_XN },
+    { "HLOS3"              ,0x86200000, 0x00B00000, AddMem, SYS_MEM, SYS_MEM_CAP, Reserv, NS_DEVICE },
+    { "TZApps"             ,0x86D00000, 0x03E00000, NoHob,  SYS_MEM, SYS_MEM_CAP, Reserv, NS_DEVICE },
+    { "TGCM"               ,0x8AB00000, 0x01400000, AddMem, SYS_MEM, SYS_MEM_CAP, Reserv, NS_DEVICE },
+    { "PIL_REGION"         ,0x8BF00000, 0x0D600000, AddMem, SYS_MEM, SYS_MEM_CAP, Reserv, NS_DEVICE },
+    { "DXE Heap"           ,0x99500000, 0x03F00000, AddMem, SYS_MEM, SYS_MEM_CAP, Conv,   WRITE_BACK_XN },
+    { "Display Reserved"   ,0x9D400000, 0x02400000, AddMem, MEM_RES, SYS_MEM_CAP, Reserv, WRITE_THROUGH },
+    { "FV Region"          ,0x9F800000, 0x00200000, AddMem, SYS_MEM, SYS_MEM_CAP, BsData, WRITE_BACK_XN },
+    { "ABOOT FV"           ,0x9FA00000, 0x00200000, AddMem, SYS_MEM, SYS_MEM_CAP, Reserv, WRITE_BACK_XN },
+    { "UEFI FD"            ,0x9FC00000, 0x00300000, AddMem, SYS_MEM, SYS_MEM_CAP, BsData, WRITE_BACK },
+    { "SEC Heap"           ,0x9FF00000, 0x0008C000, AddMem, SYS_MEM, SYS_MEM_CAP, BsData, WRITE_BACK_XN },
+    { "CPU Vectors"        ,0x9FF8C000, 0x00001000, AddMem, SYS_MEM, SYS_MEM_CAP, BsData, WRITE_BACK },
+    { "MMU PageTables"     ,0x9FF8D000, 0x00003000, AddMem, SYS_MEM, SYS_MEM_CAP, BsData, WRITE_BACK_XN },
+    { "UEFI Stack"         ,0x9FF90000, 0x00040000, AddMem, SYS_MEM, SYS_MEM_CAP, BsData, WRITE_BACK_XN },
+    { "RSRV1"              ,0x9FFD0000, 0x0000A000, AddMem, SYS_MEM, SYS_MEM_CAP, RtData, WRITE_BACK_XN },
+    { "TPMControl"         ,0x9FFDA000, 0x00003000, AddMem, MEM_RES, UNCACHEABLE, RtData, UNCACHED_UNBUFFERED_XN },
+    { "Reset Data"         ,0x9FFDD000, 0x00004000, AddMem, SYS_MEM, SYS_MEM_CAP, RtData, UNCACHED_UNBUFFERED_XN },
+    { "RSRV2"              ,0x9FFE1000, 0x00001000, AddMem, SYS_MEM, SYS_MEM_CAP, RtData, WRITE_BACK_XN },
+    { "Capsule Header"     ,0x9FFE2000, 0x00001000, AddMem, SYS_MEM, SYS_MEM_CAP, RtData, UNCACHED_UNBUFFERED_XN },
+    { "RSRV3"              ,0x9FFE3000, 0x00014000, AddMem, SYS_MEM, SYS_MEM_CAP, RtData, WRITE_BACK_XN },
+    { "Log Buffer"         ,0x9FFF7000, 0x00008000, AddMem, SYS_MEM, SYS_MEM_CAP, RtData, WRITE_BACK_XN },
+    { "Info Blk"           ,0x9FFFF000, 0x00001000, AddMem, SYS_MEM, SYS_MEM_CAP, RtData, WRITE_BACK_XN },
+
+    { "RAM Partition"      ,0xA0000000, 0x40000000, AddMem, SYS_MEM, SYS_MEM_CAP, Conv, WRITE_BACK_XN },
```

Comment:
On the sdm845 platform, we support automatic memory allocation, so please do not fix it here.